tests/Tests.elm

```diff
@@ -73,107 +78,312 @@ expectationTests =
 
-                        isNotWithin =
-                            Expect.notWithin delta a b
-                    in
-                        Expect.notEqual (succeeded isWithin) (succeeded isNotWithin)
-            , fuzz3 float float float "within commutativity" <|
-                \epsilon a b ->
-                    succeeded (Expect.within epsilon a b) |> Expect.equal (succeeded <| Expect.within epsilon b a)
-            , fuzz3 float float float "notWithin commutativity" <|
-                \epsilon a b ->
-                    succeeded (Expect.notWithin epsilon a b) |> Expect.equal (succeeded <| Expect.notWithin epsilon b a)
-            , fuzz2 float float "within reflexive" <|
-                \epsilon a ->
-                    Expect.within epsilon a a
-            , expectToFail <|
-                fuzz2 float float "notWithin reflexive" <|
+                   Despite all this effort, this will not cover all use-cases. The only way to get
+                   perfect tolerance is to analyze the floating point operations you are doing in your
+                   own program, and calculate what the maximum tolerance, relative and absolute, should
+                   be for your specific run. This also differs based on what numbers you are running
+                   through your algorithm. We do however expect this to handle all use-cases for almost
+                   all developers. If you're relying on high precision in denormalized floats, you
+                   might have to implement a solution specific to your problem.
+
+                Want we want to test:
+
+                    - some getting-up-to-speed test
+                        - pythagorean identity, pi
+
+                    - negative/positive number combinations of all below:
+                        - U near-inf
+                        - UF edge large - small
+                        - UF edge small - abs
+                        - UF edge abs - zero
+                        - UF edge abs - negative abs
+                        - U test cases in documentation
+                        - U infinity equals self
+                        - F infinity does not equal finite numbers
+                        - U NaN does not equal self
+                        - F NaN does not equal anything
+                        - F reflexivity, no matter what positive epsilon is used
+                        - U zero tolerance reflexivity
+                        - F zero tolerance does not equal anything other than self
+                        - U zero tolerance removes abs check
+                        - U zero signed/unsigned comparison
+                        - F negative tolerance matches nothing
+                        - F smoke large
+                        - F smoke small
+                        - F smoke abs
+
+                   at least 47 tests, 3 of which are for documentation and one is a demo test
+
+            -}
+            [ describe "doctests"
```

Comment:
They're manually written regression tests, to make sure that the code in the documentation for `within` does the same thing when a user runs them in the future; would be a bit embarrassing otherwise. https://github.com/drathier/elm-test/blob/master/src/Expect.elm#L248 They're also here at the beginning to ease the person reading the test cases into how `within` works in practice. 

`describe "examples in documentation"`?